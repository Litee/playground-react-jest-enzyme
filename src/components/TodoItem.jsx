import React, { Component } from 'react'

class TodoItem extends Component {
  render () {
    return (
      <li>
        {this.props.title} <button onClick={this.props.onDelete}>delete</button>
      </li>
    )
  }
}

export default TodoItem
