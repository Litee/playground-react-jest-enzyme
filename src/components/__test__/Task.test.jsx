import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import TodoItem from '../TodoItem'

test('Must render properly', () => {
  const wrapper = shallow(<TodoItem title={'Do something'} />)
  expect(shallowToJson(wrapper)).toMatchSnapshot()
})

test('Must remove todo', () => {
  const todo = 'Do something'
  const onDelete = jest.fn()
  const wrapper = shallow(<TodoItem title={todo} onDelete={onDelete.bind(this, todo)} />)

  wrapper.find('button').simulate('click')
  expect(onDelete.mock.calls.length).toBe(1)
  expect(onDelete.mock.calls[0][0]).toBe(todo)
})
