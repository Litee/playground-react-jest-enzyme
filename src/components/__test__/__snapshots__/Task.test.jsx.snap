// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Must render properly 1`] = `
<li>
  Do something
   
  <button>
    delete
  </button>
</li>
`;
